src/Parser/MarkdownParser.php

```diff
@@ -2,34 +2,42 @@
 
 declare(strict_types=1);
 
-namespace Codelicia\Xulieta\External;
+namespace Codelicia\Xulieta\Parser;
 
 use Codelicia\Xulieta\ValueObject\SampleCode;
+use Symfony\Component\Finder\SplFileInfo;
 
 use function array_pop;
 use function array_shift;
 use function count;
 use function explode;
 use function implode;
+use function in_array;
 use function ltrim;
 use function preg_match;
 use function preg_split;
 
 use const PREG_SPLIT_DELIM_CAPTURE;
 
-final class Markinho
+final class MarkdownParser implements Parser
 {
     private const PATTERN = '/\n?(`{3}[\w]*\n[\S\s]+?\n\`{3})\n/';
 
-    private function __construct()
+    public function supportedExtensions(): array
     {
+        return ['markdown', 'md'];
+    }
+
+    public function isValid(SplFileInfo $file): bool
+    {
+        return in_array($file->getExtension(), $this->supportedExtensions(), false);
```

Comment:
This should be `return true`.

A markdown file is valid even if the extension is not supported.